File: 1.1/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceSurfaceSupportKHR.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceSurfaceSupportKHR extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceSurfaceSupportKHR() {
    }

    protected PFNvkGetPhysicalDeviceSurfaceSupportKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceSurfaceSupportKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceSurfaceSupportKHR(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceSurfaceSupportKHR(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceSurfaceSupportKHR"));
    }

    public int call(VkPhysicalDevice physicalDevice, int queueFamilyIndex, VkSurfaceKHR surface, VkBool32 pSupported){
        return callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), queueFamilyIndex, surface != null ? surface.getVkAddress() : VkPointer.getNullAddressNative(), pSupported != null ? pSupported.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long physicalDevice, int queueFamilyIndex, long surface, long pSupported);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceSurfaceSupportKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceSurfaceSupportKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint queueFamilyIndex, jlong surface, jlong pSupported){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceSurfaceSupportKHR f = (PFN_vkGetPhysicalDeviceSurfaceSupportKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (uint32_t)queueFamilyIndex,
        *((VkSurfaceKHR*)jniLongToPointer(surface)),
        ((VkBool32*)jniLongToPointer(pSupported))
    );
}

